ch.tutteli.gradle.plugins.publish fails when used as a precompiled script plugin

Consider https://docs.gradle.org/current/userguide/structuring_software_products.html

Consider user creates `build-logic.published-to-central.gradle.kts` plugin:

```kotlin
plugins {
    id("ch.tutteli.gradle.plugins.dokka")
    id("ch.tutteli.gradle.plugins.publish")
}

project.extensions.getByType<ch.tutteli.gradle.plugins.publish.PublishPluginExtension>().apply {
    resetLicenses("EUPL-1.2")
}
```

Then, it fails as follows since Grade applies `build-logic.published-to-central` to a temporary `gradle-kotlin-dsl-accessors` project in order to precompile the accessors.

I suggest moving the verification to `afterEvaluate` so the users can apply `kotlin` plugin after `ch.tutteli.gradle.plugins.publish`, and I suggest skipping the verification for `gradle-kotlin-dsl-accessors`: https://github.com/robstoll/tutteli-gradle-plugins/blob/638c8c16ccb8070fcf3f9fdb46f59bfb7bd96f0b/tutteli-gradle-publish/src/main/kotlin/ch.tutteli.gradle.plugins.publish/PublishPlugin.kt#L40-L44
An alternative option could be removing the verification altogether.

```
 Caused by: org.gradle.api.internal.plugins.PluginApplicationException: Failed to apply plugin 'ch.tutteli.gradle.plugins.publish'.
                at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:173)
                at org.gradle.api.internal.plugins.DefaultPluginManager.apply(DefaultPluginManager.java:137)
                at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.lambda$applyPlugin$1(DefaultPluginRequestApplicator.java:148)
                at org.gradle.plugin.use.internal.DefaultPluginRequestApplicator.applyPlugin(DefaultPluginRequestApplicator.java:199)
                ... 133 more
                Caused by: java.lang.IllegalStateException: The project gradle-kotlin-dsl-accessors does not have any sources. We currently require a project to have sources in order to publish it.
                Please make sure you do not apply the ch.tutteli.tutteliPublish plugin before the plugin which provides the sourceSets (e.g. kotlin or java and the like)
                Please open an issue if you would like to publish projects without sources: https://github.com/robstoll/tutteli-gradle-plugins/issues/new
                    at ch.tutteli.gradle.plugins.publish.PublishPlugin.apply(PublishPlugin.kt:40)
                    at ch.tutteli.gradle.plugins.publish.PublishPlugin.apply(PublishPlugin.kt:23)
                    at org.gradle.api.internal.plugins.ImperativeOnlyPluginTarget.applyImperative(ImperativeOnlyPluginTarget.java:43)
                    at org.gradle.api.internal.plugins.RuleBasedPluginTarget.applyImperative(RuleBasedPluginTarget.java:51)
                    at org.gradle.api.internal.plugins.DefaultPluginManager.addPlugin(DefaultPluginManager.java:187)
                    at org.gradle.api.internal.plugins.DefaultPluginManager.access$100(DefaultPluginManager.java:52)
                    at org.gradle.api.internal.plugins.DefaultPluginManager$AddPluginBuildOperation.run(DefaultPluginManager.java:282)
                    at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
                    at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
                    at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
                    at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
                    at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
                    at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
                    at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
                    at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
                    at org.gradle.api.internal.plugins.DefaultPluginManager.lambda$doApply$0(DefaultPluginManager.java:167)
                    at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:44)
                    at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:166)
                    ... 136 more
```

--- a/tutteli-gradle-publish/src/main/kotlin/ch.tutteli.gradle.plugins.publish/PublishPlugin.kt
+++ b/tutteli-gradle-publish/src/main/kotlin/ch.tutteli.gradle.plugins.publish/PublishPlugin.kt
@@ -37,12 +37,6 @@ class PublishPlugin : Plugin<Project> {
         project.plugins.apply(MavenPublishPlugin::class.java)
         project.plugins.apply(SigningPlugin::class.java)
 
-        if (!project.hasProperty("sourceSets")) throw IllegalStateException(
-            "The project ${project.name} does not have any sources. We currently require a project to have sources in order to publish it." +
-                "\nPlease make sure you do not apply the ch.tutteli.$EXTENSION_NAME plugin before the plugin which provides the sourceSets (e.g. kotlin or java and the like)" +
-                "\nPlease open an issue if you would like to publish projects without sources: https://github.com/robstoll/tutteli-gradle-plugins/issues/new"
-        )
-
         val extension = project.extensions.create<PublishPluginExtension>(EXTENSION_NAME, project)
 
         val validateBeforePublish = project.tasks.register<ValidateBeforePublishTask>(TASK_NAME_VALIDATE_PUBLISH)

--- a/tutteli-gradle-publish/src/test/groovy/ch/tutteli/gradle/plugins/publish/PublishPluginValidationTest.groovy
+++ b/tutteli-gradle-publish/src/test/groovy/ch/tutteli/gradle/plugins/publish/PublishPluginValidationTest.groovy
@@ -1,8 +1,6 @@
 package ch.tutteli.gradle.plugins.publish
 
 import org.gradle.api.Project
-import org.gradle.api.internal.plugins.PluginApplicationException
-import org.gradle.testfixtures.ProjectBuilder
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.function.Executable
 
@@ -11,22 +9,6 @@ import static ch.tutteli.gradle.plugins.test.Asserts.assertThrowsProjectConfigEx
 import static org.junit.jupiter.api.Assertions.*
 
 class PublishPluginValidationTest {
-
-    @Test
-    void apply_noSourceSets_throwsIllegalStateException() {
-        //arrange
-        Project project = ProjectBuilder.builder()
-            .withName(ARTIFACT_ID)
-            .build()
-        //act && assert
-        def exception = assertThrows(PluginApplicationException) {
-            project.plugins.apply(PublishPlugin)
-        }
-        //assert
-        assertEquals(IllegalStateException, exception.cause.class)
-        def message = "The project $project.name does not have any sources"
-        assertTrue(exception.cause.message.contains(message), "cause.message contains $message:\n$exception.cause.message")
-    }
 
     @Test
     void evaluate_versionUnspecified_throwsIllegalStateException() {

